src/test/java/com/hazelcast/aws/TagsTest.java

```diff
@@ -0,0 +1,133 @@
+package com.hazelcast.aws;
+
+import com.hazelcast.aws.Tags.Tag;
+import org.junit.Test;
+
+import java.util.Iterator;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertNull;
+
+public class TagsTest {
+
+    @Test
+    public void singleTagKeyAndValue() {
+        // given
+        String tagKey = "KeyA";
+        String tagValue = "ValueA";
+
+        // when
+        Tags tags = Tags.from(tagKey, tagValue);
+
+        // then
+        Tag tag = tags.iterator().next();
+        assertEquals(tagKey, tag.getKey());
+        assertEquals(tagValue, tag.getValue());
+    }
+
+    @Test
+    public void multipleTagKeysAndValues() {
+        // given
+        String tagKey = "KeyA,KeyB";
+        String tagValue = "ValueA,ValueB";
+
+        // when
+        Tags tags = Tags.from(tagKey, tagValue);
+
+        // then
+        Iterator<Tag> iterator = tags.iterator();
+        Tag tag = iterator.next();
+        assertEquals("KeyA", tag.getKey());
+        assertEquals("ValueA", tag.getValue());
+
+        // and
+        tag = iterator.next();
+        assertEquals("KeyB", tag.getKey());
+        assertEquals("ValueB", tag.getValue());
+    }
+
+    @Test
+    public void multipleTagKeys() {
+        // given
+        String tagKey = "KeyA,KeyB";
+
+        // when
+        Tags tags = Tags.from(tagKey, null);
+
+        // then
+        Iterator<Tag> iterator = tags.iterator();
+        Tag tag = iterator.next();
+        assertEquals("KeyA", tag.getKey());
+        assertNull(tag.getValue());
+
+        // and
+        tag = iterator.next();
+        assertEquals("KeyB", tag.getKey());
+        assertNull(tag.getValue());
+    }
+
+    @Test
+    public void multipleTagValues() {
+        // given
+        String tagValue = "ValueA,ValueB";
+
+        // when
+        Tags tags = Tags.from(null, tagValue);
+
+        // then
+        Iterator<Tag> iterator = tags.iterator();
+        Tag tag = iterator.next();
+        assertNull(tag.getKey());
+        assertEquals("ValueA", tag.getValue());
+
+        // and
+        tag = iterator.next();
+        assertNull(tag.getKey());
+        assertEquals("ValueB", tag.getValue());
+    }
+
+    @Test
+    public void singleTagKeyAndMultipleTagValues() {
+        // given
+        String tagKey = "KeyA";
+        String tagValue = "ValueA,ValueB";
+
+        // when
+        Tags tags = Tags.from(tagKey, tagValue);
+
+        // then
+        Iterator<Tag> iterator = tags.iterator();
+        Tag tag = iterator.next();
+        assertEquals("KeyA", tag.getKey());
+        assertEquals("ValueA", tag.getValue());
+
+        // and
+        tag = iterator.next();
+        assertNull(tag.getKey());
+        assertEquals("ValueB", tag.getValue());
+    }
+
+    @Test
+    public void noTags() {
+        // when
```

Comment:
Nit: it would be great if you can define given/when/then steps for this test too.

Reply:
Since this test was covering to cases I have divided it into two and added `given` block for each.